--- /Users/anutosh491/miniconda3/envs/xeus-r/lib/R/include/R_ext/Complex.h.bak	2024-09-20 20:08:36
+++ /Users/anutosh491/miniconda3/envs/xeus-r/lib/R/include/R_ext/Complex.h	2024-09-20 20:11:14
@@ -73,13 +73,22 @@
 # endif
 #endif
 
+#if defined(_MSC_VER)
+// MSVC-specific definition
+typedef struct {
+    double r;
+    double i;
+} Rcomplex;
+#else
+// Definition for GCC, Clang, and other compilers
 typedef union {
     struct {
-	double r;
-	double i;
+        double r;
+        double i;
     };
     double _Complex private_data_c;
 } Rcomplex;
+#endif
 
 #ifdef __cplusplus
 # if defined(__clang__)
